apps/ejabberd/src/mod_muc_db_odbc.erl

```diff
@@ -0,0 +1,147 @@
+-module(mod_muc_db_odbc).
+-behaviour(mod_muc_db).
+-export([init/2,
+         store_room/4,
+         restore_room/3,
+         forget_room/3,
+         get_rooms/2,
+         can_use_nick/4,
+         get_nick/3,
+         set_nick/4]).
+
+-include("ejabberd.hrl").
+-include("jlib.hrl").
+-include_lib("ejabberd/include/mod_muc.hrl").
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+init(_Host, _Opts) ->
+    ok.
+
+store_room(LServer, Host, Name, Opts) ->
+    SHost = mongoose_rdbms:escape(Host),
+    SName = mongoose_rdbms:escape(Name),
+    BinOpts = jlib:term_to_expr(Opts),
+    SBinOpts = mongoose_rdbms:escape(BinOpts),
+    mongoose_rdbms:sql_transaction(
+      LServer,
+      fun() ->
+          rdbms_queries:update_t(<<"muc_room">>,
```

Comment:
The question is still valid: is there a reason why prepared statements are not used? I'm not saying it's wrong. :)